routes/accounts.js

```diff
@@ -160,152 +319,55 @@ module.exports = torIps => {
 				});
 			});
 		} else if (accountCreationDisabled.status && !hasBypass) {
-			res.status(403).json({
-				message:
-					'Creating new accounts is temporarily disabled most likely due to a spam/bot/griefing attack.  If you need an exception, please contact our moderators on discord.'
+			const creationDisabledSignup = new Signups({
+				date: new Date(),
+				userName: username,
+				type: 'Failed - ACD',
+				ip: obfIP(signupIP),
+				email: Boolean(email),
+				unobfuscatedIP: signupIP
+			});
+			creationDisabledSignup.save(() => {
+				res.status(403).json({
+					message:
+						'Creating new accounts is temporarily disabled most likely due to a spam/bot/griefing attack.  If you need an exception, please contact our moderators on discord.'
+				});
 			});
 		} else if (torIps.includes(signupIP)) {
-			const newSignup = new Signups({
+			const torSignup = new Signups({
 				date: new Date(),
 				userName: username,
-				type: 'TOR signup attempt',
-				ip: 'TOR',
-				email: Boolean(email)
+				type: 'Failed - TOR',
+				ip: obfIP(signupIP),
+				email: Boolean(email),
+				unobfuscatedIP: signupIP
 			});
-
-			newSignup.save(() => {
+			torSignup.save(() => {
 				res.status(403).json({
 					message: 'Use of TOR is not allowed on this site.'
 				});
 			});
 		} else {
-			let doesContainBadWord = false;
-
-			blacklistedWords.forEach(word => {
-				if (new RegExp(word, 'i').test(username)) {
-					doesContainBadWord = true;
-				}
-			});
-
-			if (email && !emailRegex.test(email)) {
-				res.status(401).json({
-					message: `That doesn't look like a valid email address.`
-				});
-			} else if (email && email.split('@')[1] && bannedEmails.includes(email.split('@')[1]) && process.env.NODE_ENV === 'production') {
-				res.status(401).json({
-					message: 'Only non-disposible email providers are allowed to create verified accounts.'
-				});
-			} else if (doesContainBadWord) {
-				res.status(401).json({
-					message: 'Your username contains a naughty word or part of a naughty word.'
-				});
+			if (VPNCache[signupIP]) {
+				// console.log('Running from cache:', signupIP, VPNCache[signupIP]);
+				continueSignup(req, res, username, password, email, signupIP, save, hasBypass, bypassKey, VPNCache[signupIP]);

```

Comment:
The typical best practice is if you need more than 4 arguments, make a configuration object and pass only that as the single argument.

Reply:
I didn't know that - makes sense though :D